pkg/scheduler/core/equivalence_cache.go

```diff
@@ -81,10 +79,12 @@ func (ec *EquivalenceCache) RunPredicate(
 	if err != nil {
 		return fit, reasons, err
 	}
+	ec.mu.Lock()
 	// Skip update if NodeInfo is stale.
 	if cache != nil && cache.IsUpToDate(nodeInfo) {
 		ec.updateResult(pod.GetName(), nodeInfo.Node().GetName(), predicateKey, fit, reasons, equivClassInfo.hash)
 	}
+	ec.mu.Unlock()
```

Comment:
Why don't move the lock part into`ec.updateResult()`?

Also, please update `equivalence_cache_test.go`.

Reply:
updated